src/frontend/js/components/CourseProductCourseRuns/index.spec.tsx

```diff
@@ -0,0 +1,366 @@
+          `Enrollment from ${dateFormatter.format(
+            new Date(courseRun.enrollment_start),
+          )} to ${dateFormatter.format(new Date(courseRun.enrollment_end))}`,
+        );
+
+        // - A radio input
+        screen.getByRole('radio', {
+          name: `Select course run from ${dateFormatter.format(
+            new Date(courseRun.start),
+          )} to ${dateFormatter.format(new Date(courseRun.end))}.`,
+        });
+      });
+
+      // A call to action should be displayed ...
+      // ... when no course run is selected, it should be disabled
+      let $button: HTMLButtonElement = screen.getByRole('button', {
+        name: 'Select a course run',
+      });
+      expect($button.disabled).toBe(true);
+
+      // ... once a course run is selected, it should be enabled and allows user to enroll
+      await act(async () => {
+        // - Select the first course run
+        const $radio = screen.getByRole('radio', {
+          name: `Select course run from ${dateFormatter.format(
+            new Date(courseRuns[0].start),
+          )} to ${dateFormatter.format(new Date(courseRuns[0].end))}.`,
+        });
+        fireEvent.click($radio);
+      });
+
+      $button = screen.getByRole('button', {
+        name: 'Enroll',
+      });
+      expect($button.disabled).toBe(false);
+
+      // - User clicks to enroll
+      fetchMock.resetHistory();
+      const enrollmentDeferred = new Deferred();
+      fetchMock.post('https://joanie.test/api/enrollments/', enrollmentDeferred.promise);
+
+      await act(async () => {
+        fireEvent.click($button);
+      });
+
+      // A spinner should be displayed
+      screen.getByRole('status', { name: 'Enrolling...' });
+
+      await act(async () => {
+        enrollmentDeferred.resolve(200);
+      });
+
+      const calls = fetchMock.calls();
+      expect(calls).toHaveLength(2);
+      // A request to create the enrollment should have been executed
+      expect(calls[0][0]).toBe('https://joanie.test/api/enrollments/');
+      expect(JSON.parse(fetchMock.calls()[0][1]!.body as string)).toEqual({
+        is_active: true,
+        order: order.id,
+        course_run: courseRuns[0].resource_link,
+      });
+      // Afterward a request to refresh the course should have been executed
+      expect(calls[1][0]).toBe(`https://joanie.test/api/courses/${course.code}/`);
+    });
+
+    it('does not allow to enroll if course run is not yet opened for enrollment', async () => {
+      const courseRun: CourseRun = JoanieCourseRunFactory.afterGenerate((cr: CourseRun) => ({
+        // - Course Run not yet opened for enrollment
+        ...cr,
+        enrollment_start: faker.date.future(0.25)().toISOString(),
+        enrollment_end: faker.date.future(0.5)().toISOString(),
+        start: faker.date.future(0.75)().toISOString(),
+        end: faker.date.future(1.0)().toISOString(),
+        state: {
+          priority: Priority.FUTURE_NOT_YET_OPEN,
```

Comment:
```suggestion
          priority: Priority.FUTURE_NOT_YET_OPEN,
```
as the code specifies >= Priority.FUTURE_NOT_YET_OPEN, I would recommend using a random with the different values FUTURE_CLOSED,  ONGOING_CLOSED,  ARCHIVED_CLOSED,  TO_BE_SCHEDULED,
to validate the test works in all these cases